scipy/signal/_ltisys.py

```diff
@@ -1242,6 +1242,13 @@ class StateSpace(LinearTimeInvariant):
         Sampling time [s] of the discrete-time systems. Defaults to `None`
         (continuous-time). Must be specified as a keyword argument, for
         example, ``dt=0.1``.
+    cast_to: dtype | 'floating64' | None, optional
```

Comment:
Everytime we do this, after 5-6 years it becomes an issue when something needs to change or a deprecation comes. If there is no use case for this, I'd suggest we don't bubble up this functionality to `StateSpace`. It's much easier to request from the user to use proper dtype for it. Integers to floats are expected to be done automatically elsewhere. StateSpace function API is not the right place to do data casting unless we implement the entire NumPy casting machinery.

Reply:
Removed in commit 3708d92. If needed, it can be easily added again.